### PR TITLE
reenable and fix comments

### PR DESCRIPTION
**Pull request by kkoehn**
Reenables comments on index and show of tasks

### CODE REVIEW

File: app/assets/javascripts/tasks_index.coffee [hunk 1]
```diff
@@ -88,21 +89,43 @@ intializeAdvancedFilter =->
...
     else
-      $search.css("border-bottom-left-radius","0px")
+      $search.css("border-bottom-left-radius", "0px")
       $advanced.css("border-bottom-right-radius", "0px")
       $advancedFilterActive.val(true)
       $drop.removeClass('fa-caret-down').addClass('fa-caret-up')
 
   if $('#order_param')
-    #    $('#' + order.value).addClass('active')
+#    $('#' + order.value).addClass('active')
```
Comment (MrSerth) [inline app/assets/javascripts/tasks_index.coffee:98]: Can / should we remove this line? It doesn't seem to work, currently

File: app/assets/javascripts/tasks_index.coffee [hunk 2]
```diff
@@ -88,21 +89,43 @@ intializeAdvancedFilter =->
...
     $('#order_created').addClass('active')
 
-    #  $('#order_rating').click ->
-    #    $('#order_rating').addClass('active')
-    #    $('#order_created').removeClass('active')
-    #    document.getElementById('order_param').value = 'order_rating'
+  #  $('#order_rating').click ->
+  #    $('#order_rating').addClass('active')
+  #    $('#order_created').removeClass('active')
+  #    document.getElementById('order_param').value = 'order_rating'
```
Comment (MrSerth) [inline app/assets/javascripts/tasks_index.coffee:104]: Can / should we remove these line? They don't seem to work, currently (I re-added them, of course)

**Review by MrSerth — COMMENTED** [review]
Great, some more places are working again -- I tested it locally, works fine :)

One additional question: Would you mind having a look at the create and update action of the CommentsController, please? A few days ago, I added a comment about the unused flash messages. Can you confirm that they are not working? Or should we remove them?

File: app/assets/javascripts/tasks_index.coffee (hunk 2 above)
Comment (kkoehn) [reply app/assets/javascripts/tasks_index.coffee:104]: I plan to add the rating next, so I would leave it there for now

File: app/assets/javascripts/tasks_index.coffee (hunk 1 above)
Comment (kkoehn) [reply app/assets/javascripts/tasks_index.coffee:98]: see below

File: app/assets/javascripts/tasks_index.coffee (hunk 2 above)
Comment (MrSerth) [reply app/assets/javascripts/tasks_index.coffee:104]: Alright, fine for me -- thanks for clarifying!

**Comment by MrSerth** [comment]
> > One additional question: Would you mind having a look at the create and update action of the CommentsController, please? A few days ago, I added a comment about the unused flash messages. Can you confirm that they are not working? Or should we remove them?
> 
> In my testing the flash messages actually worked. The problem is, that the flash-messages get displayed at the top of the page and you may not see them, because you are scrolled down. A fix would be to refactor the flash system....

Oh wow, thanks for making me aware! I really wondered how that would work (and already had a look) but could find out how (and as the comments didn't work; I couldn't test it). Now, I repeated my investigation and identified that the flash system uses custom headers to transport the messages 😩

https://github.com/openHPI/codeharbor/blob/9a95415ba68184a044f1389f861dd83038839626/app/controllers/application_controller.rb#L19-L26

https://github.com/openHPI/codeharbor/blob/daa46d0aa116f13d5d0c7407030d991d255e4ee6/app/assets/javascripts/flash_message.js.coffee#L7-L10

This resolves my question -- I'll drop my #ToDo note on the master branch directly. Sorry again!

**Review by MrSerth — APPROVED** [review]

**Pull request merged**